src/app/features/public-form/edit-prefilled-values/edit-prefilled-values.component.ts

```diff
@@ -0,0 +1,126 @@
+import { FormConfig } from "app/core/entity-details/form/form.component";
+import { DefaultValueConfig } from "app/core/entity/schema/default-value-config";
+import { HelpButtonComponent } from "app/core/common-components/help-button/help-button.component";
+import { MatButtonModule } from "@angular/material/button";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { FormFieldConfig } from "app/core/common-components/entity-form/FormConfig";
+import { EntityFieldSelectComponent } from "app/core/entity/entity-field-select/entity-field-select.component";
+
+@Component({
+  selector: "app-edit-prefilled-values",
+  standalone: true,
+  imports: [
+    DefaultValueOptionsComponent,
+    CommonModule,
+    ReactiveFormsModule,
+    MatFormFieldModule,
+    MatSelectModule,
+    MatTooltipModule,
+    HelpButtonComponent,
+    FontAwesomeModule,
+    MatButtonModule,
+    EntityFieldSelectComponent,
+  ],
+  templateUrl: "./edit-prefilled-values.component.html",
+  styleUrls: ["./edit-prefilled-values.component.scss"],
+})
+export class EditPrefilledValuesComponent
+  extends EditComponent<FormFieldConfig[]>
+  implements OnInit
+{
+  entityConstructor: EntityConstructor;
+  formConfig: FormConfig;
+  FormFieldConfig: FormFieldConfig;
+  defaultValue: DefaultValueConfig;
+
+  private entities = inject(EntityRegistry);
+
+  private fb = inject(FormBuilder);
+
+  prefilledValueSettings = this.fb.group({
+    prefilledValue: this.fb.array([]),
+  });
+
+  override ngOnInit(): void {
+    if (!this.entity) return;
+
+    this.entityConstructor = this.entities.get(this.entity["entity"]);
+    this.initializePrefilledValues();
+    this.prefilledValueSettings.valueChanges.subscribe((value) =>
+      this.updateFieldGroups(value),
+    );
+  }
+
+  get prefilledValues(): FormArray {
+    return this.prefilledValueSettings.get("prefilledValue") as FormArray;
+  }
+
+  private initializePrefilledValues(): void {
+    const fields = this.formControl.value as FormFieldConfig[];
+    if (!fields || !Array.isArray(fields)) return;
+
+    fields.forEach((field) => {
+      this.prefilledValues.push(
+        this.fb.group({
+          field: [field.id, Validators.required],
+          defaultValue: [field.defaultValue],
+          hideFromForm: [field.hideFromForm ?? true],
+        }),
+      );
+    });
+  }
+
+  addPrefilledFields(): void {
+    this.prefilledValues.push(
+      this.fb.group({
+        field: ["", Validators.required],
+        defaultValue: {
+          mode: "static",
+        },
+        hideFromForm: true,
+      }),
+    );
+  }
+
+  get isAddDisabled(): boolean {
+    return this.prefilledValues.invalid;
+  }
+
+  removePrefilledFields(index: number): void {
+    if (index < 0 || index >= this.prefilledValues.length) {
+      return;
+    }
+
+    this.prefilledValues.removeAt(index);
+    this.formControl.markAsDirty();
+  }
+
+  private updateFieldGroups(value): void {
```

Comment:
can you add typing for the value parameter?

I am still not fully clear on your structure here. Isn't the single property "prefilledValue" somewhat irrelevant and we could just directly put objects with id+defaultValue+hideFromForm within the formGroup and array?